.github/workflows/build.yml

```diff
@@ -26,18 +26,63 @@ jobs:
       python_version: ${{ steps.generate-matrix.outputs.python_version }}
       build: ${{ steps.generate-matrix.outputs.build}}
       test: ${{ steps.generate-matrix.outputs.test}}
+      do_macos: ${{ steps.generate-matrix.outputs.do_macos}}
+      do_ubuntu: ${{ steps.generate-matrix.outputs.do_ubuntu}}
+      do_windows: ${{ steps.generate-matrix.outputs.do_windows}}
+      build_doc: ${{ steps.generate-matrix.outputs.build_doc}}
     steps:
       - uses: actions/setup-python@v2
       - name: Generate Matrix
         id: generate-matrix
         shell: python3 {0}
         run: |
+          # Default variables
           python_version = ["3.7", "3.8", "3.9"]
-          build = { "macos": ["macos-10.15"], "linux": ["ubuntu-latest"], "windows": ["windows-2016"] }
-          test = { "macos": ["macos-10.15", "macos-11"], "linux": ["ubuntu-18.04", "ubuntu-20.04"], "windows": ["windows-2016", "windows-2019", "windows-2022"] }
+          build = [ "macos-10.15", "ubuntu-18.04", "windows-2016" ]
+          test = [ "macos-10.15", "macos-11", "ubuntu-18.04", "ubuntu-20.04", "windows-2016", "windows-2019" ]
```

Comment:
why did you remove windows-2022?  This is not mentioned in your description
BTW it builds without error.